uploader.go

```diff
@@ -29,14 +30,25 @@ type PutClient interface {
 	PutObjectWithContext(context.Context, *s3.PutObjectInput, ...request.Option) (*s3.PutObjectOutput, error)
 }
 
+type InputData interface {
+	// the actual data
+	GetBlob() []byte
+
+	// GetKey generates the key string given the time at which the first blob arrived. It is invoked
+	// immediately when a buffer is closed, and used to name/locate the object as it gets uploaded.
+	GetKey(time.Time) string
+
+	GetObjectType() string
```

Comment:
What does the object type represent?

Reply:
The non-time based portion of what `GetKey()` returns. Any recommendations on verbiage? I'll add a comment